UPGRADING.md

```diff
@@ -1,15 +1,22 @@
 # Upgrading Tendermint Core
 
-This guide provides instructions for upgrading to specific versions of Tendermint Core.
-
+This guide provides instructions for upgrading to specific versions of
+Tendermint Core.
 
 ## v0.37 (Unreleased)
 
-This version requires a coordinated network upgrade. It alters the elements in the predigest of the `LastResultsHash` and thus
-all nodes must upgrade together (see #9175).
+This version requires a coordinated network upgrade. It alters the elements in
+the predigest of the `LastResultsHash` and thus all nodes must upgrade together
+(see [\#9175](https://github.com/tendermint/tendermint/pull/9175)).
 
 NOTE: v0.35 was recalled and v0.36 was skipped
 
+### ABCI Changes
+
+* In v0.34, messages on the wire used to be length-delimited with `int64`
+  values, which was inconsistent with the `uint64` length delimiters used in the
+  P2P layer. Both now consistently use `uint64` length delimiters.
```

Comment:
I think this makes the change more confusing: because what we're doing is just using varint encoding rather than encoding (effecitively) two fixed width integers. In this statement we say "here's the thing that we changed" and not "here's what you need to do (update your socket client implementation.)" 

Reading this, my gut says "but aren't uint64 and int64 integers encoded the same on all twos-compliment machines?" and while this is true, the change is actually about varint encoding.

Reply:
We were using varint encoding before, but we were varint-encoding an `int64` value instead of a `uint64` value. The two types of varint encodings produce different values on the wire, making them incompatible.

Client libraries (such as the one in Rust) usually support the `uint64` varint encoding, and not `int64`, meaning that if anyone wants to interface with ABCI via TSP they need to roll their own length-prefixed decoding mechanism.

I'll clarify here.